tool/teleport/common/debug_test.go

```diff
@@ -0,0 +1,203 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package common
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"io"
+	"net"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/lib/config"
+	"github.com/gravitational/teleport/lib/srv/debug"
+	logutils "github.com/gravitational/teleport/lib/utils/log"
+)
+
+func TestSetLogLevel(t *testing.T) {
+	configFilePath, dataDir := newConfigWithDataDir(t)
+	closeFn := newSocketMockService(t, dataDir, []byte{})
+	defer closeFn()
+
+	// All supported log levels should be accepted here.
+	for _, level := range logutils.SupportedLevelsText {
+		t.Run(level, func(t *testing.T) {
+			err := onSetLogLevel(configFilePath, level)
+			require.NoError(t, err)
+		})
+	}
+
+	// Random or any other slog format should be rejected.
+	for _, level := range []string{"RANDOM", "DEBUG-1", "INFO+1", "INVALID"} {
+		t.Run(level, func(t *testing.T) {
+			err := onSetLogLevel(configFilePath, level)
+			require.Error(t, err)
+		})
+	}
+}
+
+func TestCollectProfiles(t *testing.T) {
+	configFilePath, dataDir := newConfigWithDataDir(t)
```

Comment:
I've updated the tests to use a different socket path per test case. The problem with this is that the UDS has a limit on the path name, and since we were using `t.TempDir()`, the path was too large. So, I had to update the listener logic to deal with it (I added a comment).

Reply:
We could. The only difference is that it will always ignore the value set to `TMPDIR` on macOS. In my tests, that doesn't seem to be a problem, but we can update it without problems. This code is the same (updated to use newer std library functions) as [`nettest` library](https://cs.opensource.google/go/x/net/+/refs/tags/v0.24.0:nettest/nettest.go;l=224).